mne/io/tests/test_read_raw.py

```diff
@@ -99,3 +99,64 @@ def test_read_raw_multiple_dots(tmp_path):
     dst = tmp_path / "test.this.file.edf"
     copyfile(src, dst)
     read_raw(dst)
+
+
+@pytest.fixture(scope="module")
+def reader_excluded_from_read_raw() -> set[str]:
+    """Set of excluded readers from read_raw."""
+    return {
+        "read_raw_bti",
+        "read_raw_hitachi",
+        "read_raw_neuralynx",
+    }
+
+
+def test_all_reader_documented(reader_excluded_from_read_raw: set[str]):
+    """Test that all the readers in the documentation are accepted by read_raw."""
+    readers = _get_supported()
+    # flatten the dictionaries and retrieve the function names
+    functions = [foo.__name__ for value in readers.values() for foo in value.values()]
+    # read documentation .rst source file
+    doc_folder = Path(__file__).parents[3] / "doc"
+    if not doc_folder.exists():
+        pytest.skip("Documentation folder not found.")
+    doc_file = doc_folder / "api" / "reading_raw_data.rst"
+    with open(doc_file) as fid:
+        doc = fid.read()
```

Comment:
More compact and let's be explicit about the encoding just to be safe (it's different on Windows and can be different on different systems so safer to always specify it knowing we stick to UTF-8)
```suggestion
    doc = doc_file.read_text("utf-8")
```